File: Basic-Math/Reverse-Bits.js
let count = 32;
let num = 12;
let reverse_num = num;
num >>= 1;
while (num) {
    reverse_num <<= 1;
    reverse_num |= num & 1;
    num >>= 1;
    count--;
}
reverse_num <<= count;

console.log(reverse_num);

File: Basic-Math/Reverse-Bits.cpp
#include <iostream>

using namespace std;

int main()
{
    cout << "Enter n:";
    int n;
    cin >> n;

    int count = 32;
    int reverseNum = n;

    do
    {
        n = n >> 1;
        reverseNum = reverseNum << 1;
        reverseNum = reverseNum | n & 1;
        count--;
    } while (n);
    reverseNum = reverseNum << count;

    cout << reverseNum;
}